packages/x-date-pickers/src/internals/hooks/usePicker/usePickerValue.ts

```diff
@@ -302,15 +302,7 @@ export const usePickerValue = <
     }
   });
 
-  if (
-    inValueWithTimezoneToRender !== undefined &&
-    (dateState.lastControlledValue === undefined ||
-      !valueManager.areValuesEqual(
-        utils,
-        dateState.lastControlledValue,
-        inValueWithTimezoneToRender,
-      ))
-  ) {
+  if (dateState.lastControlledValue !== inValueWithoutRenderTimezone) {
```

Comment:
We compare the old `props.value` with the new `props.value` (without any processing done to it)
If it changed:

1. We set the new `props.value` in the state
2. If the new derived `props.value` with the right timezone is not equal to the current draft, we update the draft